src/Raven.Client/Documents/Replication/Messages/ReplicationMessageReply.cs

```diff
@@ -13,56 +16,47 @@ internal enum ReplyType
         }
 
         public ReplyType Type { get; set; }
-
         public long LastEtagAccepted { get; set; }
-        
         public string Exception { get; set; }
-
         public string Message { get; set; }
-
         public string MessageType { get; set; }
-
         public ChangeVectorEntry[] ChangeVector { get; set; }
-
         public string DatabaseId { get; set; }
         
         public long CurrentEtag { get; set; }
     }
-
-    public static class ChangeVectorExtensions
-    {
-        public static DynamicJsonArray ToJson(this ChangeVectorEntry[] self)
-        {
-            var results = new DynamicJsonArray();
-            foreach (var entry in self)
-                results.Add(entry.ToJson());
-            return results;
-        }
-    }
-
-    public struct ChangeVectorEntry : IComparable<ChangeVectorEntry>, IDynamicJson
+    public struct ChangeVectorEntry : IComparable<ChangeVectorEntry>
     {
         public Guid DbId;
         public long Etag;
-
+        public int NodeTag;
+        public void Append(StringBuilder sb)
+        {
+            ChangeVectorExtensions.ToBase26(sb, NodeTag);
+            sb.Append(":");
+            sb.Append(Etag);
+            sb.Append("-");
+            //TODO: Fix this allocation mess
+            sb.Append(Convert.ToBase64String(DbId.ToByteArray()));
```

Comment:
We have `Base64.ConvertToBase64Array` that can probably do that without any allocations.
Note that our format calls for dropping the last few `==` used for padding.